pennylane/devices/qubit/sampling.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to sample a state."""
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+def sample_state(state, shots: int, rng: np.random.Generator = None) -> np.ndarray:
+    """
+    Returns a series of samples of a state.
+
+    Args:
+        state (array[complex]): A state vector to be sampled
+        shots (int): The number of samples to take
+        rng (Optional[np.random.Generator]): A random number generator to create samples.
+            If no RNG is provided, a default one will be used
+
+    Returns:
+        ndarray: Sample values of the shape (shots, num_wires)
+    """
+    if rng is None:
+        rng = np.random.default_rng()
```

Comment:
```suggestion
    rng = np.random.default_rng(rng)
```
If input `rng` is `None`, then it creates a new one with a random seed. If `rng` is a seed, the that is used to initialize a new `rng`. If its a generator already, then it just returns the provided generator.

This means we can always extend the input to the same set up types accepted by `default_rng`: `{None, int, array_like[ints], SeedSequence, BitGenerator, Generator}`